### PR TITLE
UHF-5047 map visibility cookie consent

### PR DESCRIPTION
**Pull request by dire**
# [UHF-5047](https://helsinkisolutionoffice.atlassian.net/browse/UHF-5047)
Maps were shown even though cookie consent was not given.

## What was done
* Display a similar cookie consent screen for maps than what videos have before the cookies are accepted.
* To reduce duplicate code, a common template was created to show cookie blocker in embedded medias
* CSS/template changes had to be done after adding the cookie blocker.

## How to install

* Make sure any instance is up and running on latest dev branch.
  * `git pull origin dev`
* Update the HDBT theme
  * `composer require drupal/hdbt:dev-UHF-5047_Map-visibility-cookie-consent`
* Run `make drush-cr`

## How to test

* [x] Add a map component to any page.
* [x] See that the map is not shown unless cookies are accepted.
* [x] Check that the map host is shown properly in the text on the blocker screen.
* [x] Check that the styling is still same for the map component.
* [x] Since the templates for the remote videos are affected, check that the video component works still fine.

## Designers review

* [x] This PR does not need designers review
* [ ] This PR has been visually reviewed by a designer (Name of the designer)

[UHF-5047]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-5047?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: templates/misc/embedded-content-cookie-compliance.twig [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<div class="embedded-content-cookie-compliance media-{{ media_id }}">
+  <div class="message">
+    {% include '@hdbt/misc/icon.twig' ignore missing with {icon: 'alert-circle-fill'} %}
+    <h2>{{ 'Content cannot be displayed'|t }}</h2>
+    <p>
+      {% trans %}
+        This content is hosted by {{ media_service_url }}. To see the content, switch over to the external site or modify your cookie settings to allow for preference and statistics cookies.
```
[the remark this answers is not in the document]
Comment (Arkkimaagi) [reply templates/misc/embedded-content-cookie-compliance.twig:7]: The variable name change breaks also the video cookie consent translation.

File: templates/misc/embedded-content-cookie-compliance.twig [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        'class': [
+          'hds-button',
+          'hds-button--secondary',
+        ],
+      } %}
+      {{ link(link_title, privacy_policy_url, link_attributes) }}
+
+    </div>
+  </div>
+</div>
```
Comment (Arkkimaagi) [inline templates/misc/embedded-content-cookie-compliance.twig:38]: Missing newline at the end of the file, please check your editor settings.

**Review by Arkkimaagi — CHANGES_REQUESTED** [review]
Almost. :)

Please see my comments below.

This branch is missing the changed dist folder files. Please create them before merging.

File: templates/misc/embedded-content-cookie-compliance.twig (hunk 1 above)
[the remark this answers is not in the document]
Comment (dire) [reply templates/misc/embedded-content-cookie-compliance.twig:7]: Changed the variable name in the fi.po and sv.po files.
Need to use `drush locale:check && drush locale:update` and probably `drush cr` to get the translations work.

File: src/js/embedded-content-cookie-compliance.js [hunk 3]
```diff
@@ -22,13 +22,18 @@
...
 
         const containerElement = document.createElement('div');
-        containerElement.classList.add('responsive-video-container');
         containerElement.appendChild(iframeElement);
-
-        $(`.embedded-content-cookie-compliance.media-${id}`)
+        if (attributes.type === 'video') {
+          containerElement.classList.add('responsive-video-container'); 
+          $(`.embedded-content-cookie-compliance.media-${id}`)
           .empty()
```
[the remark this answers is not in the document]
Comment (dire) [reply src/js/embedded-content-cookie-compliance.js]: Fixed!

File: src/js/embedded-content-cookie-compliance.js [hunk 4]
```diff
@@ -22,13 +22,18 @@
...
           .append(containerElement)
           .removeClass(`media-${id}`);
+        } else if (attributes.type === 'map') {
+          containerElement.classList.add('responsive-map-container');
+          $(`.embedded-content-cookie-compliance.media-${id}`)
+          .replaceWith(containerElement);
```
[the remark this answers is not in the document]
Comment (dire) [reply src/js/embedded-content-cookie-compliance.js]: Fixed!

File: src/js/embedded-content-cookie-compliance.js [hunk 5]
```diff
@@ -22,13 +22,18 @@
         iframeElement.title = attributes.title;
 
         const containerElement = document.createElement('div');
-        containerElement.classList.add('responsive-video-container');
         containerElement.appendChild(iframeElement);
-
-        $(`.embedded-content-cookie-compliance.media-${id}`)
+        if (attributes.type === 'video') {
```
[the remark this answers is not in the document]
Comment (dire) [reply src/js/embedded-content-cookie-compliance.js:26]: Agh, I did but forgot to commit adding the media type in the `media--remote-video.html.twig`🤦‍♂️. Should be fixed now!

File: templates/misc/embedded-content-cookie-compliance.twig (hunk 2 above)
Comment (dire) [reply templates/misc/embedded-content-cookie-compliance.twig]: Fixed and checked!

**Comment by dire** [comment]
Good catches! All should be fixed now.

File: templates/misc/embedded-content-cookie-compliance.twig (hunk 1 above)
[the remark this answers is not in the document]
Comment (Arkkimaagi) [reply templates/misc/embedded-content-cookie-compliance.twig:7]: Works

File: src/js/embedded-content-cookie-compliance.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (Arkkimaagi) [reply src/js/embedded-content-cookie-compliance.js:26]: Works now :)

**Review by Arkkimaagi — APPROVED** [review]
Works now, good job!

**Pull request merged**